### PR TITLE
Add user notes

### PR DESCRIPTION
**Pull request by WallE256**
Implements #768

### CODE REVIEW

File: judge/migrations/0075_add_user_notes.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        ('judge', '0074_totp'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='profile',
+            name='notes',
+            field=models.TextField(blank=True, help_text='Notes for administrators regarding this user.', null=True, verbose_name='internal notes'),
+        ),
+        migrations.AlterField(
```
Comment (Xyene) [inline judge/migrations/0075_add_user_notes.py]: This shouldn't be in this patch

File: judge/migrations/0075_add_user_notes.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            model_name='comment',
+            name='page',
+            field=models.CharField(db_index=True, max_length=30, validators=[django.core.validators.RegexValidator(b'^[pcs]:[a-z0-9]+$|^b:\\d+$', 'Page code must be ^[pcs]:[a-z0-9]+$|^b:\\d+$')], verbose_name='associated page'),
+        ),
+        migrations.AlterField(
```
Comment (Xyene) [inline judge/migrations/0075_add_user_notes.py]: This shouldn't be in this patch

File: judge/migrations/0075_add_user_notes.py (hunk 1 above)
Comment (quantum5) [reply judge/migrations/0075_add_user_notes.py]: Frankly I attribute this to django generating `AlterField` non-database changes. But in any case, the migration needs to be synced eventually. The convention, however ugly, is to sneak it in with the next actual database change. We should probably work out a better system.

**Comment by kiritofeng** [comment]
**Previews:**

Admin interface:
image
As superuser:
image
Impersonating a non-staff user:
image

**Pull request merged**